xarray/core/dataarray.py

```diff
@@ -761,7 +761,7 @@ def data(self, value: Any) -> None:
     @property
     def values(self) -> np.ndarray:
         """
-        The array's data as a numpy.ndarray.
+        A view onto the DataArray's underlying data as a numpy.ndarray.
```

Comment:
(I'm no expert here.)

Is this correct? I would have thought that it's returning the ndarray. And then that can get used as a view in many situations, consistent with numpy's rules.

If my sense is correct, we could just comment that it's not copied, and so mutations to the array will also affect the dataarray?

Reply:
Thanks for the clarification! How's this? 

"The array's underlying `numpy.ndarray`. Note that this array is not copied; operations on it follow `numpy`'s rules of what generates a view vs. a copy, and changes to this array will be reflected in the DataArray as well."

Reply:
Perfect I think!

(though to confirm, I'm not confident that the initial thing isn't so accurate, fine to wait for others to comment if we prefer...)

Reply:
what we're doing is [roughly](https://github.com/pydata/xarray/blob/4806412c7f08480d283f1565262f040bdc74d11d/xarray/core/variable.py#L292-L312)
```python
return np.asarray(self._data)
```
(with slightly different behavior for 0d arrays with `datetime64` or `timedelta64` dtype)

This means that "The array's data converted to numpy.ndarray." would be a more accurate summary, but we could definitely emphasize the forwarding towards `numpy.asarray` / `numpy.array` a bit more (the current "If the array's data is not a numpy.ndarray this will attempt to convert it naively using np.array()" is wrong, `numpy.asarray` is *always* called).

Reply:
Great, thanks.

So let's merge after this is done...

Reply:
How's this? 
"The array's data converted to numpy.ndarray. Note that this array is not copied; operations on it follow numpy's rules of what generates a view vs. a copy, and changes to this array may be reflected in the DataArray as well."

Reply:
if there's two line breaks between the first and second sentences I don't have any objections (the first line is supposed to be a brief summary). Just make sure to coordinate the stuff that's already there:

>  If the array's data is not a numpy.ndarray this will attempt to convert it naively using np.array(), which will raise an error if the array type does not support coercion like this (e.g. cupy).

Reply:
Sounds good - from your comment above, should "convert it naively using np.array()" just be changed to "convert it naively using np.asarray()" then? Or should that clause just be struck

Reply:
I'm not sure about the difference between `asarray` and `array` (I *think* it's different defaults), but in any case the important part is that the function is always applied to the function, not just when it's not already a `numpy` array.

Reply:
Ohhh okay - makes sense. I think the tweaks in the latest push should address that